lua/nvim-treesitter/configs.lua

```diff
@@ -4,76 +4,95 @@ local queries = require'nvim-treesitter.query'
 local parsers = require'nvim-treesitter.parsers'
 local utils = require'nvim-treesitter.utils'
 
--- @enable can be true or false
--- @disable is a list of languages, only relevant if enable is true
--- @keymaps list of user mappings for a given module if relevant
--- @is_supported function which, given a ft, will return true if the ft works on the module
 local config = {
```

Comment:
@vigoux aren't those locals not overwritten on every require. Also before it was not possible to get the current config after a user call of `setup` since it was always set back to the default values. `:lua print(vim.inspect(require'nvim-treesitter.configs'.get_module('incremental_selection') ` returns `nil`

Reply:
I'll look into that. Does lua run the file on every require or is it cached after the first require of it?

Reply:
Also, this depends when you get the module. The built-in modules don't get defined until `require('nvim-treesitter').setup()` is called.

Reply:
I guess lua runs the whole file on every require. You can check that by placing a small print statement next to the local definition. It is at least called again when a user runs it. Don't whether it's cached when the same thread requires it again.

Reply:
@theHamsta What's the desired result? If I run `:lua print(require'nvim-treesitter.configs'.get_module('incremental_selection'))` I get a table. Lua doesn't run the file for each require (I tried what you suggested with the print statements) so the config state shouldn't be affected.

I think the one key difference with this implementation vs the previous is modules are now defined async. They could be defined at any time. So you can't "get all modules", since a plugin may not have defined it yet, but you can get all registered at that time.

Reply:
Nice! Now its also working for me. Thanks for checking!